gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineServiceQueryParser.java

```diff
@@ -0,0 +1,219 @@
+                    QueryRequestUtil.createFilter(
+                        entityIdAttribute,
+                        Operator.IN,
+                        createStringArrayLiteralExpression(entityIds)))
+            .collect(Collectors.toList()));
+
+    // Time range is a mandatory filter for query service, hence add it if it's not already present.
+    filterBuilder.addChildFilter(
+        QueryRequestUtil.createBetweenTimesFilter(timeColumn, startTimeInMillis, endTimeInMillis));
+
+    return filterBuilder;
+  }
+
+  public BaselineEntitiesResponse parseQueryResponse(
+      Iterator<ResultSetChunk> resultSetChunkIterator,
+      BaselineRequestContext requestContext,
+      int idColumnsSize,
+      String entityType,
+      long periodSecs,
+      long startTime,
+      long endTime) {
+    Map<String, AttributeMetadata> attributeMetadataMap =
+        attributeMetadataProvider.getAttributesMetadata(requestContext, entityType);
+    Map<EntityKey, Map<String, BaselineMetricSeries.Builder>> entityMetricSeriesMap =
+        new LinkedHashMap<>();
+    while (resultSetChunkIterator.hasNext()) {
+      ResultSetChunk chunk = resultSetChunkIterator.next();
+      LOG.debug("Received chunk: {} ", chunk);
+
+      if (chunk.getRowCount() < 1) {
+        break;
+      }
+
+      for (Row row : chunk.getRowList()) {
+        EntityKey entityKey =
+            EntityKey.of(
+                IntStream.range(0, idColumnsSize)
+                    .mapToObj(value -> row.getColumn(value).getString())
+                    .toArray(String[]::new));
+
+        Map<String, BaselineMetricSeries.Builder> metricSeriesMap =
+            entityMetricSeriesMap.computeIfAbsent(entityKey, k -> new LinkedHashMap<>());
+
+        BaselineInterval.Builder intervalBuilder = BaselineInterval.newBuilder();
+
+        // Second column is the time column
+        Value value =
+            QueryAndGatewayDtoConverter.convertQueryValueToGatewayValue(
+                row.getColumn(idColumnsSize));
+        if (value.getValueType() == ValueType.STRING) {
+          long time = Long.parseLong(value.getString());
+          intervalBuilder.setStartTimeMillis(time);
+          intervalBuilder.setEndTimeMillis(time + TimeUnit.SECONDS.toMillis(periodSecs));
+
+          for (int i = idColumnsSize + 1;
+              i < chunk.getResultSetMetadata().getColumnMetadataCount();
+              i++) {
+            ColumnMetadata metadata = chunk.getResultSetMetadata().getColumnMetadata(i);
+            BaselineTimeAggregation timeAggregation =
+                requestContext.getTimeAggregationByAlias(metadata.getColumnName());
+
+            if (timeAggregation == null) {
+              LOG.warn("Couldn't find an aggregate for column: {}", metadata.getColumnName());
+              continue;
+            }
+
+            Value convertedValue;
+            // AVG_RATE is adding a specific implementation because Pinot does not directly support this function,
```

Comment:
Move comment to where the method is implemented

Reply:
Done